### PR TITLE
Feature/public api startup check

### PR DESCRIPTION
**Pull request by IdoZilberberg**
e2e now starts immediately when all docker containers are healthy; added startup check for Public API Service.

### CODE REVIEW

File: e2e/src/test-environment/orbs-nodes.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class OrbsNode implements TestComponent {
       command += " --forceRecreate";
     }
     await this.runDockerCompose(command);
-    await delay(60000);
+    // await delay(60000);
```
Comment (jlevison) [inline e2e/src/test-environment/orbs-nodes.ts]: why was this removed? if not required remove comment

File: projects/services/public-api-service-typescript/src/service.ts [hunk 2]
```diff
@@ -43,6 +52,25 @@ export default class PublicApiHTTPService extends Service {
...
+      startupCheckRunner.run()
+        .then((result: StartupStatus) => {
+          const httpCode = result.status === STARTUP_STATUS.OK ? 200 : 503;
+          return res.status(httpCode).send(result);
+        });
+    });
+
+    const mgmtServer = mgmtApp.listen(httpManagementPort);
+    logger.info(`Started HTTP public api management server on port ${httpManagementPort}`);
+    return mgmtServer;
```
Comment (jlevison) [inline projects/services/public-api-service-typescript/src/service.ts]: it checks that the public api 'business logic' is running, but can it also check if the express started okay and running? (meaning the service itself)

File: e2e/src/test-environment/orbs-nodes.ts (hunk 1 above)
Comment (IdoZilberberg) [reply e2e/src/test-environment/orbs-nodes.ts]: Replaced by actual logic that proceeds with the test once all docker containers are healthy, instead of waiting arbitrary time. Comment removed.

File: projects/services/public-api-service-typescript/src/service.ts (hunk 2 above)
Comment (IdoZilberberg) [reply projects/services/public-api-service-typescript/src/service.ts]: Let's talk on Sunday about it. Maybe it's worth adding to the other services too.

File: e2e/docker-containers-health-check.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
```
Comment (netoneko) [inline e2e/docker-containers-health-check.sh]: Is there any reason to use this script instead of writing it in TypeScript? With TS it would be much more maintainable.

File: projects/services/public-api-service-typescript/src/service.ts (hunk 2 above)
Comment (IdoZilberberg) [reply projects/services/public-api-service-typescript/src/service.ts]: Added check that Public API HTTP Server is alive and listening.

File: e2e/docker-containers-health-check.sh (hunk 3 above)
Comment (IdoZilberberg) [reply e2e/docker-containers-health-check.sh]: It's not the concern of the TS code; also easier to test this code from command line when it's a shell script.

Comment (IdoZilberberg) [reply e2e/docker-containers-health-check.sh]: Replaced shell script code with ts code for docker healthcheck.
